### ISSUE TITLE
SC2119 / SC2120 false positives?

### ISSUE THREAD

**Issue by dubo-dubon-duponey** [koalaman/shellcheck#2926]
#### For bugs
- Rule Id (if any, e.g. SC1000):  SC2119 / SC2120
- My shellcheck version (`shellcheck --version` or "online"): online and 0.9.0
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash

foo(){
        local input="${1:-/dev/stdin}"
        cat "$input"
}


foo <<<"blah"
```

#### Here's what shellcheck currently says:

```
In test.sh line 3:
foo(){
^-- SC2120 (warning): foo references arguments, but none are ever passed.


In test.sh line 9:
foo <<<"blah"
^-----------^ SC2119 (info): Use foo "$@" if function's $1 should mean script's $1.

For more information:
  https://www.shellcheck.net/wiki/SC2120 -- foo references arguments, but non...
  https://www.shellcheck.net/wiki/SC2119 -- Use foo "$@" if function's $1 sho...
```

#### Here's what I wanted or expected to see:

~~It should say nothing about SC2120. An argument *is* being passed (a file descriptor), so, this one feels factually wrong to me.~~

SC2119, I am not quite sure what it would mean here, so, same thing, I would expect it to not say anything.

Apologies in advance if I am missing something here.

Thanks a lot!

**Comment by nhed** [comment]
I agree, IMO shellcheck should ignore this if any of the following are used - but especially the first

```
       ${parameter:-word}
       ${parameter:=word}
       ${parameter:?word}
       ${parameter:+word}
```

**Comment by qneill-sifive** [comment]
Related to or maybe a duplicate of https://github.com/koalaman/shellcheck/issues/2511

**Comment by koalaman** [comment]
This has been fixed in #2023 for `:-`. The others (aside from `:=` which doesn't work on positional parameters) are included in 553a80f77ad. Thanks!
